src/app/libraries/http_lib.php

```diff
@@ -1,14 +1,19 @@
 <?php
 
 class http_lib extends Library {
-
     function response_code($code, $view_and_exit = true, $view_name = false) {
         http_response_code($code);
         if ($view_and_exit) {
             if (empty($view_name)) {
                 $view_name = strval($code);
             }
-            $this->load_view($view_name);
+            // Check if a dedicated view exists; if not, load common error view
+            $dedicated_view = $this->load_view($view_name);
```

Comment:
We should pass status code because here we can give view name directly, and that view can be a shared view for more than one errors. It is not always dedicated view.